@davet2001 - Sure thing, I updated the new tests. I was copying from plenty of other tests that only look for `state is not None`.

Regarding the rest of the `unittest` tests - I made naive changes to pytest style, but it times out locally:

```python
async def test_uptime_sensor_days_output(hass):
    """Test uptime sensor output data."""
    sensor = UptimeSensor("test", "days")
    assert sensor.unit_of_measurement == "days"
    new_time = sensor.initial + timedelta(days=1)
    with patch("homeassistant.util.dt.now", return_value=new_time):
        await    asyncio.run_coroutine_threadsafe(
            sensor.async_update(), hass.loop
        ).result()
        assert sensor.state == 1.00
    new_time = sensor.initial + timedelta(days=111.499)
    with patch("homeassistant.util.dt.now", return_value=new_time):
        await asyncio.run_coroutine_threadsafe(
            sensor.async_update(), hass.loop
        ).result()
        assert sensor.state == 111.50

```

yields:

```shell
$ $ pytest --timeout=9 --durations=10 -n auto --dist=loadfile -qq -o console_output_style=count -p no:sugar --cov-report term-missing --cov=homeassistant.components.uptime.sensor tests/components/uptime/test_sensor.py
[SNIP]
=========================================================== FAILURES ===========================================================
____________________________________________ test_uptime_sensor_days_output[pyloop] ____________________________________________
[gw0] linux -- Python 3.7.5 /core/venv/bin/python3.7

hass = <homeassistant.core.HomeAssistant object at 0x7f91316dac90>

    async def test_uptime_sensor_days_output(hass):
        """Test uptime sensor output data."""
        sensor = UptimeSensor("test", "days")
        assert sensor.unit_of_measurement == "days"
        new_time = sensor.initial + timedelta(days=1)
        with patch("homeassistant.util.dt.now", return_value=new_time):
            await    asyncio.run_coroutine_threadsafe(
>               sensor.async_update(), hass.loop
            ).result()

tests/components/uptime/test_sensor.py:56: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.7/concurrent/futures/_base.py:430: in result
    self._condition.wait(timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.7/concurrent/futures/_base.py:430: in result
    self._condition.wait(timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Condition(<unlocked _thread.RLock object owner=0 count=0 at 0x7f91316accc0>, 0)>, timeout = None

    def wait(self, timeout=None):
[SNIP]
            if timeout is None:
>               waiter.acquire()
E               Failed: Timeout >9.0s

/usr/lib/python3.7/threading.py:296: Failed

```

--- a/tests/components/uptime/test_sensor.py
+++ b/tests/components/uptime/test_sensor.py
@@ -1,5 +1,4 @@
 """The tests for the uptime sensor platform."""
-import asyncio
 from datetime import timedelta
 
 from homeassistant.components.uptime.sensor import UptimeSensor
@@ -50,15 +49,11 @@ async def test_uptime_sensor_days_output(hass):
     assert sensor.unit_of_measurement == "days"
     new_time = sensor.initial + timedelta(days=1)
     with patch("homeassistant.util.dt.now", return_value=new_time):
-        await asyncio.run_coroutine_threadsafe(
-            sensor.async_update(), hass.loop
-        ).result()
+        await sensor.async_update()
         assert sensor.state == 1.00
     new_time = sensor.initial + timedelta(days=111.499)
     with patch("homeassistant.util.dt.now", return_value=new_time):
-        await asyncio.run_coroutine_threadsafe(
-            sensor.async_update(), hass.loop
-        ).result()
+        await sensor.async_update()
         assert sensor.state == 111.50
 
 
@@ -68,15 +63,11 @@ async def test_uptime_sensor_hours_output(hass):
     assert sensor.unit_of_measurement == "hours"
     new_time = sensor.initial + timedelta(hours=16)
     with patch("homeassistant.util.dt.now", return_value=new_time):
-        await asyncio.run_coroutine_threadsafe(
-            sensor.async_update(), hass.loop
-        ).result()
+        await sensor.async_update()
         assert sensor.state == 16.00
     new_time = sensor.initial + timedelta(hours=72.499)
     with patch("homeassistant.util.dt.now", return_value=new_time):
-        await asyncio.run_coroutine_threadsafe(
-            sensor.async_update(), hass.loop
-        ).result()
+        await sensor.async_update()
         assert sensor.state == 72.50
 
 
@@ -86,13 +77,9 @@ async def test_uptime_sensor_minutes_output(hass):
     assert sensor.unit_of_measurement == "minutes"
     new_time = sensor.initial + timedelta(minutes=16)
     with patch("homeassistant.util.dt.now", return_value=new_time):
-        await asyncio.run_coroutine_threadsafe(
-            sensor.async_update(), hass.loop
-        ).result()
+        await sensor.async_update()
         assert sensor.state == 16.00
     new_time = sensor.initial + timedelta(minutes=12.499)
     with patch("homeassistant.util.dt.now", return_value=new_time):
-        await asyncio.run_coroutine_threadsafe(
-            sensor.async_update(), hass.loop
-        ).result()
+        await sensor.async_update()
         assert sensor.state == 12.50

--- a/tests/components/uptime/test_sensor.py
+++ b/tests/components/uptime/test_sensor.py
@@ -1,13 +1,11 @@
 """The tests for the uptime sensor platform."""
 import asyncio
 from datetime import timedelta
-import unittest
 
 from homeassistant.components.uptime.sensor import UptimeSensor
 from homeassistant.setup import async_setup_component
 
 from tests.async_mock import patch
-from tests.common import get_test_home_assistant
 
 
 async def test_uptime_min_config(hass):
@@ -46,61 +44,55 @@ async def test_uptime_sensor_config_minutes(hass):
     assert state.attributes.get("unit_of_measurement") == "minutes"
 
 
-class TestUptimeSensor(unittest.TestCase):
-    """Test the uptime sensor."""
+async def test_uptime_sensor_days_output(hass):
+    """Test uptime sensor output data."""
+    sensor = UptimeSensor("test", "days")
+    assert sensor.unit_of_measurement == "days"
+    new_time = sensor.initial + timedelta(days=1)
+    with patch("homeassistant.util.dt.now", return_value=new_time):
+        await asyncio.run_coroutine_threadsafe(
+            sensor.async_update(), hass.loop
+        ).result()
+        assert sensor.state == 1.00
+    new_time = sensor.initial + timedelta(days=111.499)
+    with patch("homeassistant.util.dt.now", return_value=new_time):
+        await asyncio.run_coroutine_threadsafe(
+            sensor.async_update(), hass.loop
+        ).result()
+        assert sensor.state == 111.50
 
-    def setUp(self):
-        """Set up things to run when tests begin."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.hass.stop)
 
-    def test_uptime_sensor_days_output(self):
-        """Test uptime sensor output data."""
-        sensor = UptimeSensor("test", "days")
-        assert sensor.unit_of_measurement == "days"
-        new_time = sensor.initial + timedelta(days=1)
-        with patch("homeassistant.util.dt.now", return_value=new_time):
-            asyncio.run_coroutine_threadsafe(
-                sensor.async_update(), self.hass.loop
-            ).result()
-            assert sensor.state == 1.00
-        new_time = sensor.initial + timedelta(days=111.499)
-        with patch("homeassistant.util.dt.now", return_value=new_time):
-            asyncio.run_coroutine_threadsafe(
-                sensor.async_update(), self.hass.loop
-            ).result()
-            assert sensor.state == 111.50
+async def test_uptime_sensor_hours_output(hass):
+    """Test uptime sensor output data."""
+    sensor = UptimeSensor("test", "hours")
+    assert sensor.unit_of_measurement == "hours"
+    new_time = sensor.initial + timedelta(hours=16)
+    with patch("homeassistant.util.dt.now", return_value=new_time):
+        await asyncio.run_coroutine_threadsafe(
+            sensor.async_update(), hass.loop
+        ).result()
+        assert sensor.state == 16.00
+    new_time = sensor.initial + timedelta(hours=72.499)
+    with patch("homeassistant.util.dt.now", return_value=new_time):
+        await asyncio.run_coroutine_threadsafe(
+            sensor.async_update(), hass.loop
+        ).result()
+        assert sensor.state == 72.50
 
-    def test_uptime_sensor_hours_output(self):
-        """Test uptime sensor output data."""
-        sensor = UptimeSensor("test", "hours")
-        assert sensor.unit_of_measurement == "hours"
-        new_time = sensor.initial + timedelta(hours=16)
-        with patch("homeassistant.util.dt.now", return_value=new_time):
-            asyncio.run_coroutine_threadsafe(
-                sensor.async_update(), self.hass.loop
-            ).result()
-            assert sensor.state == 16.00
-        new_time = sensor.initial + timedelta(hours=72.499)
-        with patch("homeassistant.util.dt.now", return_value=new_time):
-            asyncio.run_coroutine_threadsafe(
-                sensor.async_update(), self.hass.loop
-            ).result()
-            assert sensor.state == 72.50
 
-    def test_uptime_sensor_minutes_output(self):
-        """Test uptime sensor output data."""
-        sensor = UptimeSensor("test", "minutes")
-        assert sensor.unit_of_measurement == "minutes"
-        new_time = sensor.initial + timedelta(minutes=16)
-        with patch("homeassistant.util.dt.now", return_value=new_time):
-            asyncio.run_coroutine_threadsafe(
-                sensor.async_update(), self.hass.loop
-            ).result()
-            assert sensor.state == 16.00
-        new_time = sensor.initial + timedelta(minutes=12.499)
-        with patch("homeassistant.util.dt.now", return_value=new_time):
-            asyncio.run_coroutine_threadsafe(
-                sensor.async_update(), self.hass.loop
-            ).result()
-            assert sensor.state == 12.50
+async def test_uptime_sensor_minutes_output(hass):
+    """Test uptime sensor output data."""
+    sensor = UptimeSensor("test", "minutes")
+    assert sensor.unit_of_measurement == "minutes"
+    new_time = sensor.initial + timedelta(minutes=16)
+    with patch("homeassistant.util.dt.now", return_value=new_time):
+        await asyncio.run_coroutine_threadsafe(
+            sensor.async_update(), hass.loop
+        ).result()
+        assert sensor.state == 16.00
+    new_time = sensor.initial + timedelta(minutes=12.499)
+    with patch("homeassistant.util.dt.now", return_value=new_time):
+        await asyncio.run_coroutine_threadsafe(
+            sensor.async_update(), hass.loop
+        ).result()
+        assert sensor.state == 12.50

